portfolio/src/main/webapp/script.js

```diff
@@ -42,3 +42,515 @@ function visitLink(s){
+            }
+            return false;
+        }
+        else if(selected_piece.type == 'king'){
+            if((square.row == this.selected_square.row + 1 || 
+                square.row == this.selected_square.row - 1 || 
+                square.col == this.selected_square.col + 1 || 
+                square.col == this.selected_square.col - 1) &&
+                square.row - this.selected_square.row < 2 &&
+                square.col - this.selected_square.col < 2 &&
+                square.row - this.selected_square.row > -2 &&
+                square.col - this.selected_square.col > -2){
+                
+                selected_piece.moved = true;
+                return true;
+            }
+            else if(square.row == this.selected_square.row && 
+                square.col == this.selected_col + 2){
+                if(selected_piece.isWhite){
+                    if(!this.row[7][5].piece && !this.row[7][6].piece){
+                        selected_piece.moved = true;
+                        return 2;
+                    }
+                }
+                else{
+                    if(!this.row[0][5].piece && !this.row[0][6].piece){
+                        selected_piece.moved = true;
+                        return 3;
+                    }
+                }
+                return false;
+            }
+            else if(square.row == this.selected_square.row && 
+                square.col == this.selected_col + 2){
+                if(selected_piece.isWhite){
+                    if(!this.row[7][1].piece && !this.row[7][2].piece && !this.row[7][3].piece){
+                        selected_piece.moved = true;
+                        return 2;
+                    }
+                }
+                else{
+                    if(!this.row[0][1].piece && !this.row[0][2].piece && !this.row[0][3].piece){
+                        selected_piece.moved = true;
+                        return 3;
+                    }
+                }
+                return false;
+            }
+            return false;
+        }
+        else{
+            return false;
+        }
+    }
+
+    squareClicked(square){
+        // alert(`${square.row} ${square.col}`);
+
+        //if there is a previously selected square
+        if(this.selected_square){
+
+            //determine if move is legal
+            let legal = this.checkLegal(square, this.selected_square.piece);
+
+            //if the move is legal
+            if(legal == true){
+                //if there is a piece on the square
+                if(square.piece){
+                    //if the piece is the same color as the piece on the previously selected square
+                    if(square.piece.isWhite == this.selected_square.piece.isWhite){
+                        this.selected_square = square;
+                        console.log("Selected piece on: (" + square.row + ", " + square.col + ")");
+                        return;
+                    }
```

Comment:
The way this is written now, since moved is set to true for legal moves of the pawn, king, and rook. If you have a white piece next to a white king and try to click the king then click the white piece, it will switch to selecting the other white piece, but the king will be marked as move. Maybe change the setting of move to part of squareClicked or move the check of if thee selected_square has another piece of the same color to before checkLegal is called.

Reply:
That's a good point. I'll look into that when I implement castling. will make a TODO comment